src/platform_impl/macos/view.rs

```diff
@@ -801,17 +813,19 @@ impl WinitView {
         }
     }
 
-    fn window(&self) -> Id<WinitWindow, Shared> {
+    fn window(&self) -> Option<Id<WinitWindow, Shared>> {
         // TODO: Simply use `window` property on `NSView`.
         // That only returns a window _after_ the view has been attached though!
         // (which is incompatible with `frameDidChange:`)
         //
         // unsafe { msg_send_id![self, window] }
-        self._ns_window.load().expect("view to have a window")
+        self._ns_window.load()
     }
 
     fn window_id(&self) -> WindowId {
-        WindowId(self.window().id())
+        self.window()
+            .map(|window| WindowId(window.id()))
+            .unwrap_or(unsafe { WindowId::dummy() })
```

Comment:
Though convenient, `WindowId::dummy` is clearly unsound, and we will have to handle this differently.